ngraph/frontend/paddlepaddle/include/paddlepaddle_frontend/model.hpp

```diff
@@ -18,28 +18,35 @@
 
 #include <frontend_manager/frontend_manager.hpp>
 #include <fstream>
+#include <string>
 
 #include "place.hpp"
 
 namespace ngraph {
 namespace frontend {
 
-class FrontEndPDPD;
-
 class NGRAPH_API InputModelPDPD : public InputModel
 {
-    // TODO: replace it by already deserialized proto hidden under some Impl class
-    // TODO: avoid using explicit format-dependent data stuctures here, hide it under some Impl class
-   
     friend class FrontEndPDPD;
+    class InputModelPDPDImpl;
+    std::shared_ptr<InputModelPDPDImpl> _impl;
```

Comment:
consider to use std::unique_ptr as pointer to private implementation as InputModelPDPD definitely has the ownership of _impl

Reply:
In that case we need to delete copy constructor for the model, right?

Reply:
yes, copy constructor will be deleted by default (as there is no copy constructor for unique_ptr)
Having copy constructor sharing the same internal implementation looks a bit confusing for me.
E.g. 
` InputModel model1 = ....;
InputModel model2 = model1;
model1->overrideAllInputs(...)
// Now model2 has also all in inputs overridden, which is not obvious and probably this is not what user of this class expects
`